mmdeploy/mmcv/ops/nms_match.py

```diff
@@ -0,0 +1,232 @@
+                    score_threshold)
+
+
+def _select_nms_index(scores: torch.Tensor,
+                      boxes: torch.Tensor,
+                      nms_index,
+                      batch_size: int,
+                      keep_top_k: int = -1):
+    """Transform NMS output.
+
+    Args:
+        scores (Tensor): The detection scores of shape
+            [N, num_classes, num_boxes].
+        boxes (Tensor): The bounding boxes of shape [N, num_boxes, 4].
+        nms_index (Tensor): NMS output of bounding boxes indexing.
+        here is [K, ?]
+        batch_size (int): Batch size of the input image.
+        keep_top_k (int): Number of top K boxes to keep after nms.
+            Defaults to -1.
+
+
+    Returns:
+        tuple[Tensor, Tensor]: (dets, labels), `dets` of shape [N, num_det, 5]
+            and `labels` of shape [N, num_det].
+    """
+    '''
+    batch_inds = torch.tensor([int(nms[0]) for nms in nms_index])
+    cls_inds = torch.tensor([0 * len(nms_index)])
+    box_inds = torch.tensor([int(nms[2]) for nms in nms_index])
+
+    '''
+    batch_inds, cls_inds = nms_index[:, 0], nms_index[:, 1]
+    box_inds = nms_index[:, 2]
+    print(batch_inds, cls_inds, box_inds)
+    # index by nms output
+    scores = scores[batch_inds, cls_inds, box_inds].unsqueeze(1)
+    boxes = boxes[batch_inds, box_inds, ...]
+    dets = torch.cat([boxes, scores], dim=1)
+
+    # batch all
+    batched_dets = dets.unsqueeze(0).repeat(batch_size, 1, 1)
+    # print(batch_inds)
+    batch_template = torch.arange(
+        0, batch_size, dtype=batch_inds.dtype, device=batch_inds.device)
+    batched_dets = batched_dets.where(
+        (batch_inds == batch_template.unsqueeze(1)).unsqueeze(-1),
+        batched_dets.new_zeros(1))
+
+    batched_labels = cls_inds.unsqueeze(0).repeat(batch_size, 1)
+    batched_labels = batched_labels.where(
+        (batch_inds == batch_template.unsqueeze(1)),
+        batched_labels.new_ones(1) * -1)
+
+    N = batched_dets.shape[0]
+
+    # expand tensor to eliminate [0, ...] tensor
+    batched_dets = torch.cat((batched_dets, batched_dets.new_zeros((N, 1, 5))),
+                             1)
+    batched_labels = torch.cat((batched_labels, batched_labels.new_zeros(
+        (N, 1))), 1)
+    # sort
+    is_use_topk = keep_top_k > 0 and \
+        (torch.onnx.is_in_onnx_export() or keep_top_k < batched_dets.shape[1])
+    if is_use_topk:
+        _, topk_inds = batched_dets[:, :, -1].topk(keep_top_k, dim=1)
+    else:
+        _, topk_inds = batched_dets[:, :, -1].sort(dim=1, descending=True)
+    topk_batch_inds = torch.arange(
+        batch_size, dtype=topk_inds.dtype,
+        device=topk_inds.device).view(-1, 1)
+    batched_dets = batched_dets[topk_batch_inds, topk_inds, ...]
+    batched_labels = batched_labels[topk_batch_inds, topk_inds, ...]
+    # slice and recover the tensor
+    return batched_dets, batched_labels
+
+
+def _multiclass_nms_match(boxes: Tensor,
```

Comment:
has this function been used?

Reply:
yes I use it on func multiclass_nms_match